validation/poststart.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	tap "github.com/mndrix/tap-go"
+	rspec "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/opencontainers/runtime-tools/specerror"
+	"github.com/opencontainers/runtime-tools/validation/util"
+	uuid "github.com/satori/go.uuid"
+)
+
+func main() {
+	t := tap.New()
+	t.Header(0)
+
+	g := util.GetDefaultGenerator()
+
+	var output string
+	config := util.LifecycleConfig{
+		Actions: util.LifecycleActionCreate | util.LifecycleActionStart | util.LifecycleActionDelete,
+		PreCreate: func(r *util.Runtime) error {
+			r.SetID(uuid.NewV4().String())
+
+			output = filepath.Join(r.BundleDir, g.Spec().Root.Path, "output")
+			poststart := rspec.Hook{
+				Path: fmt.Sprintf("%s/%s/bin/sh", r.BundleDir, g.Spec().Root.Path),
+				Args: []string{
+					"sh", "-c", fmt.Sprintf("echo 'post-start called' >> %s", output),
+				},
+			}
+
+			g.AddPostStartHook(poststart)
+			g.SetProcessArgs([]string{"sh", "-c", fmt.Sprintf("echo 'process called' >> %s", output)})
+			r.SetConfig(g)
+			return nil
+		},
+		PostCreate: func(r *util.Runtime) error {
+			outputData, err := ioutil.ReadFile(output)
+			if err == nil {
+				if strings.Contains(string(outputData), "post-start called") {
+					return specerror.NewError(specerror.PoststartTiming, fmt.Errorf("The post-start hooks MUST be called before the `start` operation returns"), rspec.Version)
+				} else if strings.Contains(string(outputData), "process called") {
+					return specerror.NewError(specerror.ProcNotRunAtResRequest, fmt.Errorf("The user-specified program (from process) MUST NOT be run at this time"), rspec.Version)
+				}
+				return fmt.Errorf("File %v should not exist", output)
+			}
+			return nil
+		},
+		PreDelete: func(r *util.Runtime) error {
```

Comment:
I suggest we can add
```
util.WaitingForStatus(*r, util.LifecycleStatusStopped, time.Second * 10, time.Second)
```
 'start' will return immediately, although 'echo' runs fast, but we are not sure how many time it costs. (for example, in runc/cc/Kata project)